matminer/featurizers/utils/roost.py

```diff
@@ -0,0 +1,308 @@
+import functools
+
+import numpy as np
+from matminer.utils.data import MatscholarElementData
+
+try:
+    import torch
+    from torch.utils.data import Dataset
+    from torch.nn import L1Loss, MSELoss, CrossEntropyLoss
+
+    from roost.utils import Normalizer
+    from roost.roost.model import Roost
+except ImportError:
+    torch, Dataset = None, object
+    L1Loss, MSELoss, CrossEntropyLoss = None, None, None
+    Normalizer, Roost = object, object
+
+
+class CompositionData(Dataset):
+    """
+    The CompositionData dataset is a wrapper for a dataset data points are
+    automatically constructed from composition strings.
+    """
+
+    def __init__(self, comp, targets=None, task="regression", emb="matscholar"):
+        """
+        """
+        self.comp = comp
+
+        if targets is not None:
+            self.targets = targets
+        else:
+            self.targets = np.full_like(comp, np.nan).ravel()
+
+        if emb == "matscholar":
+            self.elem_features = MatscholarElementData()
+        else:
+            raise NotImplementedError(
+                "Currently only the matscholar embedding is implemented"
+            )
+
+        self.task = task
+        if self.task == "regression":
+            self.n_targets = 1
+        elif self.task == "classification":
+            if targets is not None:
+                self.n_targets = np.max(self.targets) + 1
```

Comment:
pytorch losses for classification take label encoded classes. I am not sure what the standard approach for `matminer` is, if it's one-hot then I need to add something to convert between the two